### ISSUE TITLE
can't deploy at local kubernetes (unbound immediate PersistentVolumeClaims)

### ISSUE THREAD

**Issue by nikosece** [mongodb/mongodb-kubernetes-operator#173]
I have a cluster with 2 physical machines and no cloud providers. i am trying to deploy  the statefulset as i want to create a mongodb with persistent. Due to local storage i have created manually some Persistent Volumes, and i am trying to bound at them with volume claim templates. here is my extra code:
`volumeClaimTemplates:
        - metadata:
            name: mongo-persistent-storage
          spec:
            accessModes: [ "ReadWriteOnce" ]
            resources:
              requests:
                storage: 20Gi
              storageClassName: "localstorage"
`
Are there specific instructions for local storages and manualy persistent volumes?
for better undestanding here are my storage yaml, and mongo deployment yaml
[mongo.zip](https://github.com/mongodb/mongodb-kubernetes-operator/files/5171759/mongo.zip)

**Comment by bznein** [comment]
Hello @nikosece and thanks for your patience.

We had some bugs around merging custom specs for volumes claims which we hopefully fixed with release 0.2.1, which is after your post. Can you please try again with the latest release and let me know if the issue persists?

Thanks

**Comment by TannerBragg** [comment]
Hi @bznein 

I just happened to give the operator a try today and observed the same behavior via the 0.2.0 operator version.  I updated the operator to 0.2.1 and it appears to still exhibit the following event when creating the first pod

`error while running "VolumeBinding" filter plugin for pod "example-openshift-mongodb-0": pod has unbound immediate PersistentVolumeClaims`

I did add the following to the custom resource replica set and it appears it mapped the bound volume to the 
```
volumeClaimTemplates:
        - metadata:
            name: mongo-persistent-storage
          spec:
            storageClassName: "fast"
            accessModes: ["ReadWriteOnce"]
            resources:
              requests:
                storage: 100Gi
```

Looking at the pod yaml, I can see the volume is properly specified that was created using the defined storage class above.  I do not see any references to `example-openshift-mongodb-0` in the pod yaml itself.  Why would it be trying to bind this volume? Lastly, looking at the `example-openshift-mongodb-0` PVC, it does not have a PV for the PVC to operate on.

```
volumes:
    - name: data-volume
      persistentVolumeClaim:
        claimName: data-volume-example-openshift-mongodb-0
    - name: mongo-persistent-storage
      persistentVolumeClaim:
        claimName: mongo-persistent-storage-example-openshift-mongodb-0
```

**Comment by TannerBragg** [comment]
@chatton @bznein In order to work around the issue, we went ahead and assigned a storage class, but it still created the failing default PVC.  Finally, we tried naming our storage class defined PVC the same name as the default to override the values and it appears to work.  

It appears when a volume claim template is specified, it tries to merge the volume claim templates rather than replace with our defined values.  So to confirm, you are not merging the values properly for storage.

It's probably not reliable long term.
